website/docs/r/data_factory_linked_custom_service.html.markdown

```diff
@@ -0,0 +1,113 @@
+---
+subcategory: "Data Factory"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_data_factory_linked_custom_service"
+description: |-
+  Manages a Linked Service (connection) between a resource and Azure Data Factory. This is a generic resource that supports all different Linked Service Type.
+---
+
+# azurerm_data_factory_linked_custom_service
+
+Manages a Linked Service (connection) between a resource and Azure Data Factory. This is a generic resource that supports all different Linked Service Type.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_data_factory" "example" {
+  name                = "example"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_storage_account" "example" {
+  name                     = "example"
+  resource_group_name      = azurerm_resource_group.example.name
+  location                 = azurerm_resource_group.example.location
+  account_kind             = "BlobStorage"
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_data_factory_linked_custom_service" "example" {
+  name               = "example"
+  data_factory_id    = azurerm_data_factory.example.id
+  type               = "AzureBlobStorage"
+  description        = "test description"
+  type_property_json = <<JSON
+{
+  "connectionString":"${azurerm_storage_account.test.primary_connection_string}"
+}
+JSON
+
+  parameters = {
+    "foo" : "bar"
+    "Env" : "Test"
+  }
+
+  annotations = [
+    "test1",
+    "test2",
+    "test3"
+  ]
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) Specifies the name of the Data Factory Linked Service. Changing this forces a new resource to be created. Must be unique within a data
+  factory. See the [Microsoft documentation](https://docs.microsoft.com/en-us/azure/data-factory/naming-rules) for all restrictions.
+
+* `data_factory_id` - (Required) The Data Factory ID in which to associate the Linked Service with. Changing this forces a new resource.
+
+* `type` - (Required) The type of data stores that will be connected to Data Factory. For full list of supported data stores, please refer to [Azure Data Factory connector](https://docs.microsoft.com/en-us/azure/data-factory/connector-overview).
+
+* `type_property_json` - (Optional) A JSON object that contains the properties of the Data Factory Linked Service.
+
+* `additional_properties` - (Optional) A map of additional properties to associate with the Data Factory Linked Service.
```

Comment:
can we sort the optional properties alphabetically?

Reply:
have made `type_properties_json` required.